### PR TITLE
remove `SubqueryFilterToJoin`

### PR DESCRIPTION
**Pull request by jackwener**
# Which issue does this PR close?

N/A

# Rationale for this change

From  comment of @liukun4515 https://github.com/apache/arrow-datafusion/pull/4725#issuecomment-1364999536

`SubqueryFilterToJoin` is useless, because `DecorrelateWhereIn` exist

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

None

### CODE REVIEW

File: datafusion/optimizer/src/decorrelate_where_in.rs [hunk 1]
```diff
@@ -220,10 +220,196 @@ mod tests {
...
+        let plan = LogicalPlanBuilder::from(table_scan)
+            .filter(and(
+                in_subquery(col("c"), test_subquery_with_name("sq_1")?),
+                in_subquery(col("b"), test_subquery_with_name("sq_2")?),
+            ))?
+            .project(vec![col("test.b")])?
+            .build()?;
+
+        let expected = "Projection: test.b [b:UInt32]\
+        \n  LeftSemi Join: test.b = __sq_2.c [a:UInt32, b:UInt32, c:UInt32]\
```
Comment (alamb) [inline datafusion/optimizer/src/decorrelate_where_in.rs:259]: nice 👍

**Review by alamb — APPROVED** [review]
Thank you @jackwener  -- this is a nice cleanup!

**Review by liukun4515 — APPROVED** [review]

**Comment by liukun4515** [comment]
cc @avantgardnerio   @korowa the author of these two rules

**Comment by alamb** [comment]
I plan to merge this tomorrow unless anyone else would like time to comment

**Comment by alamb** [comment]
Thanks @jackwener and @liukun4515

**Pull request merged**